cmd/kubeadm/app/cmd/alpha/certs.go

```diff
@@ -89,14 +89,20 @@ func getRenewSubCommands() []*cobra.Command {
 	kubeadmutil.CheckErr(err)
 
 	cmdList := []*cobra.Command{}
-	allCmds := []func() error{}
+	funcList := []func(){}
 
 	for caCert, certs := range certTree {
 		// Don't offer to renew CAs; would cause serious consequences
 		for _, cert := range certs {
-			cmd := makeCommandForRenew(cert, caCert, cfg)
+			// get the cobra.Command skeleton for this command
+			cmd := generateRenewalCommand(cert, cfg)
+			// get the implementation of renewing this certificate
+			renewalFunc := generateRenewalFunction(cert, caCert, cfg)
+			// install the implementation into the command
+			cmd.Run = func(*cobra.Command, []string) { renewalFunc() }
```

Comment:
could generateRenewalFunction() return a function with the `func(*cobra.Command, []string)` signature?